File: src/Nebula/Modules/Cashier/Models/CajaDiaria.cs
using Nebula.Modules.Account.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Nebula.Modules.Cashier.Models;

public class CajaDiaria
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>
    /// Identificador de la empresa al que pertenece.
    /// </summary>
    public Guid? CompanyId { get; set; } = null;

    [ForeignKey(nameof(CompanyId))]
    public Company Company { get; set; } = new Company();

    /// <summary>
    /// ID Serie de facturación.
    /// </summary>
    public Guid? InvoiceSerieId { get; set; } = null;

    [ForeignKey(nameof(InvoiceSerieId))]
    public InvoiceSerie InvoiceSerie { get; set; } = new InvoiceSerie();

    /// <summary>
    /// Series de facturación.
    /// </summary>
    public string Terminal { get; set; } = string.Empty;

    /// <summary>
    /// Estado Caja (ABIERTO|CERRADO).
    /// </summary>
    public string Status { get; set; } = string.Empty;

    /// <summary>
    /// Monto Apertura.
    /// </summary>
    public decimal TotalApertura { get; set; }

    /// <summary>
    /// Monto para el dia siguiente.
    /// </summary>
    public decimal TotalCierre { get; set; }

    /// <summary>
    /// Turno Operación de caja.
    /// </summary>
    public string Turno { get; set; } = string.Empty;

    /// <summary>
    /// Fecha de Operación.
    /// </summary>
    public string CreatedAt { get; set; } = DateTime.Now.ToString("yyyy-MM-dd");

    /// <summary>
    /// Año de registro.
    /// </summary>
    public string Year { get; set; } = DateTime.Now.ToString("yyyy");

    /// <summary>
    /// Mes de registro.
    /// </summary>
    public string Month { get; set; } = DateTime.Now.ToString("MM");

    /// <summary>
    /// Identificador del Almacén.
    /// Esta propiedad se usa solo para mostrar datos.
    /// </summary>
    public Guid? WarehouseId { get; set; } = null;

    [ForeignKey(nameof(WarehouseId))]
    public Warehouse Warehouse { get; set; } = new Warehouse();
}

File: ngx-angular/src/app/cashier/interfaces/caja-diaria.ts
import {Company} from "../../account/interfaces";
import {Contact} from "../../contact/interfaces";

export class CajaDiaria {
  constructor(
    public id: any = undefined,
    public companyId: string = "",
    public invoiceSerieId: string = "", // Id Serie de facturación.
    public terminal: string = "", // identificador y nombre - Serie de facturación.
    public status: string = "", // Estado de Caja.
    public totalApertura: number = 0, // Monto Apertura.
    public totalCierre: number = 0, // Monto para el dia siguiente.
    public turno: string = "", // turno del trabajador.
    public createdAt: any = undefined,
    public warehouseId: any = undefined,) {
  }
}

export class QuickSaleConfig {
  company: Company = new Company();
  cajaDiaria: CajaDiaria = new CajaDiaria();
  contact: Contact = new Contact();
}

export interface AperturaCaja {
  serieId: string;
  total: number;
  turno: string;
}

export interface CerrarCaja {
  totalCierre: number;
}

export class ResumenCajaDto {
  constructor(
    public yape: number = 0,
    public credito: number = 0,
    public contado: number = 0,
    public deposito: number = 0,
    public salida: number = 0,
    public montoTotal: number = 0) {
  }
}
